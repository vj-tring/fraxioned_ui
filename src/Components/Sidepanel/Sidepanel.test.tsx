import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router, MemoryRouter } from 'react-router-dom';
import SidePanel from './Sidepanel';

jest.mock('../../assets/images/fraxioned_logo.png', () => 'fraxioned_logo.png');
jest.mock('../../assets/Fraxioned-icon.png', () => 'fraxioned_icon.png');

describe('SidePanel', () => {
    const mockToggleSidebar = jest.fn();

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('renders the component correctly', () => {
        render(
            <Router>
                <SidePanel isOpen={true} toggleSidebar={mockToggleSidebar} />
            </Router>
        );
    });

    test('toggles the sidebar when the logo is clicked', () => {
        render(
            <Router>
                <SidePanel isOpen={false} toggleSidebar={mockToggleSidebar} />
            </Router>
        );

        fireEvent.click(screen.getByAltText('Fraxioned Owners\' Portal'));
        expect(mockToggleSidebar).toHaveBeenCalled();
    });

    test('toggles the sidebar when the toggle button is clicked', () => {
        render(
            <Router>
                <SidePanel isOpen={false} toggleSidebar={mockToggleSidebar} />
            </Router>
        );

        fireEvent.click(screen.getByRole('button'));
        expect(mockToggleSidebar).toHaveBeenCalled();
    });

    test('disables menu items when they are marked as disabled', () => {
        render(
            <MemoryRouter initialEntries={['/admin/bookings']}>
                <SidePanel isOpen={true} toggleSidebar={mockToggleSidebar} />
            </MemoryRouter>
        );

        const disabledLink = screen.getByText('Holidays');
        fireEvent.click(disabledLink);
        expect(mockToggleSidebar).not.toHaveBeenCalled();
    });

    test('highlights the active menu item', () => {
        render(
            <MemoryRouter initialEntries={['/admin/bookings']}>
                <SidePanel isOpen={true} toggleSidebar={mockToggleSidebar} />
            </MemoryRouter>
        );
    });

    test('matches the snapshot', () => {
        const { container } = render(
            <Router>
                <SidePanel isOpen={true} toggleSidebar={mockToggleSidebar} />
            </Router>
        );
        expect(container).toMatchSnapshot();
    });
});