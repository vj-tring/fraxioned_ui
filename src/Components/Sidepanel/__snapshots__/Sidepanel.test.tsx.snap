// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel matches the snapshot 1`] = `
<div>
  <nav
    class="sidePanel open"
  >
    <div
      class="logoContainer"
    >
      <img
        alt="Fraxioned Owners' Portal"
        class="logo"
        src="fraxioned_logo.png"
      />
      <button
        class="toggleButton"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </button>
    </div>
    <ul
      class="menu"
    >
      <li
        class="menuItem"
      >
        <a
          class="menuLink  "
          href="/admin/bookings"
        >
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
              />
            </svg>
          </span>
          <span
            class="label"
          >
            Bookings
          </span>
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="menuLink  disabled"
          href="/holidays"
        >
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M480 192H365.71L260.61 8.06A16.014 16.014 0 0 0 246.71 0h-65.5c-10.63 0-18.3 10.17-15.38 20.39L214.86 192H112l-43.2-57.6c-3.02-4.03-7.77-6.4-12.8-6.4H16.01C5.6 128-2.04 137.78.49 147.88L32 256 .49 364.12C-2.04 374.22 5.6 384 16.01 384H56c5.04 0 9.78-2.37 12.8-6.4L112 320h102.86l-49.03 171.6c-2.92 10.22 4.75 20.4 15.38 20.4h65.5c5.74 0 11.04-3.08 13.89-8.06L365.71 320H480c35.35 0 96-28.65 96-64s-60.65-64-96-64z"
              />
            </svg>
          </span>
          <span
            class="label"
          >
            Holidays
          </span>
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="menuLink  disabled"
          href="/user"
        >
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              />
            </svg>
          </span>
          <span
            class="label"
          >
            User
          </span>
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="menuLink  disabled"
          href="/documents"
        >
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 384 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              />
            </svg>
          </span>
          <span
            class="label"
          >
            Documents
          </span>
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="menuLink  disabled"
          href="/role"
        >
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 640 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M630.6 364.9l-90.3-90.2c-12-12-28.3-18.7-45.3-18.7h-79.3c-17.7 0-32 14.3-32 32v79.2c0 17 6.7 33.2 18.7 45.2l90.3 90.2c12.5 12.5 32.8 12.5 45.3 0l92.5-92.5c12.6-12.5 12.6-32.7.1-45.2zm-182.8-21c-13.3 0-24-10.7-24-24s10.7-24 24-24 24 10.7 24 24c0 13.2-10.7 24-24 24zm-223.8-88c70.7 0 128-57.3 128-128C352 57.3 294.7 0 224 0S96 57.3 96 128c0 70.6 57.3 127.9 128 127.9zm127.8 111.2V294c-12.2-3.6-24.9-6.2-38.2-6.2h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 287.9 0 348.1 0 422.3v41.6c0 26.5 21.5 48 48 48h352c15.5 0 29.1-7.5 37.9-18.9l-58-58c-18.1-18.1-28.1-42.2-28.1-67.9z"
              />
            </svg>
          </span>
          <span
            class="label"
          >
            Role
          </span>
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="menuLink  disabled"
          href="/reports"
        >
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M332.8 320h38.4c6.4 0 12.8-6.4 12.8-12.8V172.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V76.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-288 0h38.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zM496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
              />
            </svg>
          </span>
          <span
            class="label"
          >
            Reports
          </span>
        </a>
      </li>
      <li
        class="menuItem"
      >
        <a
          class="menuLink  disabled"
          href="/rules"
        >
          <span
            class="icon"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504.971 199.362l-22.627-22.627c-9.373-9.373-24.569-9.373-33.941 0l-5.657 5.657L329.608 69.255l5.657-5.657c9.373-9.373 9.373-24.569 0-33.941L312.638 7.029c-9.373-9.373-24.569-9.373-33.941 0L154.246 131.48c-9.373 9.373-9.373 24.569 0 33.941l22.627 22.627c9.373 9.373 24.569 9.373 33.941 0l5.657-5.657 39.598 39.598-81.04 81.04-5.657-5.657c-12.497-12.497-32.758-12.497-45.255 0L9.373 412.118c-12.497 12.497-12.497 32.758 0 45.255l45.255 45.255c12.497 12.497 32.758 12.497 45.255 0l114.745-114.745c12.497-12.497 12.497-32.758 0-45.255l-5.657-5.657 81.04-81.04 39.598 39.598-5.657 5.657c-9.373 9.373-9.373 24.569 0 33.941l22.627 22.627c9.373 9.373 24.569 9.373 33.941 0l124.451-124.451c9.372-9.372 9.372-24.568 0-33.941z"
              />
            </svg>
          </span>
          <span
            class="label"
          >
            Rules
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
